util.go

```diff
@@ -107,3 +107,17 @@ func condWrite(w io.Writer, repeat int, runes ...rune) error {
 	_, err := w.Write(buf)
 	return err
 }
+
+// repeat b c times and write it to w, returning b with increased capacity
+func repeat(w io.Writer, b []byte, c int) []byte {
+	if cap(b) >= c {
```

Comment:
This assumes that the whole underlying array is filled with same bytes, which doesn't have to be true. We should only rely on length, not capacity since we always write a subslice anyway.

I might not have seen any issues here where running tests because we don't expect them to pass, which will be costly in the long run.

Because of this, I converted this PR to a draft.